compiler/rustc_borrowck/src/type_check/free_region_relations.rs

```diff
@@ -133,7 +133,8 @@ impl UniversalRegionRelations<'_> {
         assert!(self.universal_regions.is_universal_region(fr0));
 
         let mut external_parents = vec![];
-        let mut queue = vec![fr0];
+
+        let mut queue = vec![relation.scc_representative(fr0)];
```

Comment:
I'm not sure if you've explained why projecting the free region to its representative affects the set of `non_local_bounds` that we return. It's not obvious from the definition of `TransitiveRelation::parents`, at least.

Like, if these are actually SCC's then I'm not sure why shuffling around which element in that SCC fixes things.

Is there something subtle going on with the fact that `'static` is the first first region vid we allocate?

Reply:
If it has to do with the fact that `TransitiveRelation::parents` requires for us to pass the *minimal* element for this while loop to actually reach all of the parents, then perhaps we should rename thatfunction to `minimal_scc_representative` 🤔

 Or maybe I'm totally misleading myself.

Reply:
Hey @matthewjasper, did you mean to resolve this without actually answering the question?

Reply:
Ah ok thanks